config/resources/domain.json

```diff
@@ -275,6 +281,11 @@
           "predicate": "prov:derivedFrom",
           "target": "document-container",
           "cardinality": "one"
+        },
+        "snippet-list": {
```

Comment:
so if I understood you correctly, you had to add this because `publishedSnippetContainer->DocumentContainer(the snippet)->SnippetList` (using the inverse of hasSnippets with cardinality 1) doesn`t work because documentcontainer is not public right?

Reply:
after talking with Aad it seems we can make specific predicates public while hiding the rest, I'll look into whether this can help us simplify some things

Reply:
I couldn't spend as much time as I hoped on this, but from talks with Aad and the docs it looks like this should work to avoid the extra link:
```elixir
defmodule Acl.UserGroups.Config do
  def user_groups do
    # These elements are walked from top to bottom.  Each of them may
    # alter the quads to which the current query applies.  Quads are
    # represented in three sections: current_source_quads,
    # removed_source_quads, new_quads.  The quads may be calculated in
    # many ways.  The useage of a GroupSpec and GraphCleanup are
    # common.
    [
      # // PUBLIC
      %GroupSpec{
        name: "public",
        useage: [:read],
        access: %AlwaysAccessible{},
        graphs: [ %GraphSpec{
                    graph: "http://mu.semte.ch/graphs/public",
                    constraint: %ResourceConstraint{
                      resource_types: [
                        "http://xmlns.com/foaf/0.1/OnlineAccount", # added here to fetch them for mock-login
                        "http://xmlns.com/foaf/0.1/Person",  # added here to fetch them for mock-login
                        "http://data.vlaanderen.be/ns/besluit#Bestuurseenheid",
                        "http://www.w3.org/2004/02/skos/core#Concept",
                        "http://www.w3.org/2004/02/skos/core#ConceptScheme",
                        "http://data.lblod.info/vocabularies/mobiliteit/Codelist",
                        "http://mu.semte.ch/vocabularies/ext/PublishedRegulatoryAttachmentContainer",
                        "http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#FileDataObject",
                        "http://mu.semte.ch/vocabularies/ext/SnippetList",
                        "http://mu.semte.ch/vocabularies/ext/PublishedSnippetContainer",
                        "http://mu.semte.ch/vocabularies/ext/PublishedSnippet",
                      ]
                    } },
                  %GraphSpec{
                    graph: "http://mu.semte.ch/graphs/public"
                    constraint: %ResourceConstraint{
                      resource_types: [
                        "http://mu.semte.ch/vocabularies/ext/DocumentContainer",
                      ],
                      predicates: %NoPredicates {
                        except: ["http://mu.semte.ch/vocabularies/ext/hasSnippetList"]
                      }
                    }

                  }

                ]
      },
```
(its just the addition of the second graphspec)
UNTESTED: beware of typos, wrong uris, and syntax errors if you copypaste

the [docs](https://github.com/mu-semtech/mu-authorization) are actually quite clear for a core service, I definitely recommend a full read-through.
see also the convo in the mu-semtech channel where I was able to verify my interpretation of the docs with Aad. 

One unanswered question with this setup is whether the `<container-uri> a <ContainerTypeUri>` triple would also end up in the public graph, since it seems the type predicate is handled in a special way

I suspect it will, and I have no idea if that's good or bad, so just keep an eye out.
One way I could see it throwing things off is if we would do a count of containers in the public graph somewhere. We should be getting 0 now, but with this change we might get the either a full count, or just the amount of containers with that specific link (which you could consider as "leaking" information :thinking: but ok). But I don't think we do that anyway.

This would also need a migration for prod to get that triple in the public graph for the existing templates (of which there is still only 1 afaik, so not really a big deal yet)

Reply:
Addendum: youd need to add the same extra config to the "authenticated-write" group as well